### DISCUSSION TITLE
Return "errorDetails": "Invalid login credentials.",

### DISCUSSION THREAD

**Discussion by shadowoff09** [pocketbase/pocketbase#1633] [Q&A] [answered]
Hi devs, is there a way to return the error details the same way we return the error.message?

**Comment by ganigeorgiev** [comment, marked as answer]
"errorDetails: Invalid login credentials" indicates missing identity or invalid password.
We can't return it in the response because `errorDetails` is the raw/plain serialized error and could contain sensitive information. It is logged only for debug purposes.

The associated auth error message ("Failed to authenticate") is deliberately generic to help preventing accounts enumeration (you can also check the owasp recommendation - https://cheatsheetseries.owasp.org/cheatsheets/Authentication_Cheat_Sheet.html#authentication-and-error-messages).

I'll consider in the next release to provide separate debug error messages for the missing `identity` and `password` validation errors, but still this will be accessible only in the logs or when in `--debug` mode.

**Reply by YoRHa-A5** [reply]
What about the response though? I am currently trying PB with the Qwik framework and trying to build an equivalent like the hooks.server.ts in SvelteKit, however I run into the issue that invalid credentials (user not in db) still results in a 200 POST request, so I can't differentiate/intercept to handle a request for a user that is not in the db. Any ideas?

**Reply by ganigeorgiev** [reply]
@BunnyTheLifeguard 

> invalid credentials (user not in db) still results in a 200 POST

Do you mean the List/Search collection endpoint? We return 200 with empty paginated `items` response if the client request doesn't satisfy the rule because the API rules act also as filters and there is no easy way to identify which exact part of the rule resulted to `false`.

If that's not the case, please file a new discussion with more details.

**Reply by YoRHa-A5** [reply]
Oh mb, I mean a simple authWithPassword, like this:

```ts
export const useLogin = routeAction$(
  async (data, { fail, redirect, status }) => {
    const authData = await pb
      .collection('users')
      .authWithPassword(data.email, data.password);

   // Does not reach the code below if authWithPassword() is used with user data that is not present in db

    if (pb.authStore.isValid) {
      redirect(303, '/');
      return {
        success: true,
      };
    } else {
      pb.authStore.clear();
      status(400);
      return fail(400, { message: 'Invalid user data' });
    }
  }
);
```

If it's getting too offtopic sure I can file a new discussion :)

**Reply by ganigeorgiev** [reply]
@BunnyTheLifeguard `authWithPassword` will return a 400 response and throws an error on invalid credentials, not 200. If you want to return your `fail(400, ...)` call, you could try wrapping the auth call in a `try/catch` block.

**Reply by YoRHa-A5** [reply]
Omg, how did I forget about try/catch? :D

Thank you my friend and happy easter to you! 🐰
